smqtk_classifier/impls/classify_descriptor_supervised/sklearn_svm.py

```diff
@@ -0,0 +1,449 @@
+                    self.svm_type = 'one-class SVM'
+                elif isinstance(self.svm_model, svm.SVR):
+                    self.svm_type = 'epsilon-SVR'
+                elif isinstance(self.svm_model, svm.NuSVR):
+                    self.svm_type = 'nu-SVR'
+                else:
+                    raise RuntimeError("Invalid SVM model type loaded.")
+
+            finally:
+                os.remove(fp)
+        else:
+            self.svm_model = None
+            self._reload_model()
+
+    def _reload_model(self) -> None:
+        """
+        Reload SVM model from configured file path.
+        """
+        if self.svm_model_elem and not self.svm_model_elem.is_empty():
+            svm_model_tmp_fp = self.svm_model_elem.write_temp()
+            self.svm_model = joblib.load(svm_model_tmp_fp)
+            self.svm_model_elem.clean_temp()
+
+        if self.svm_label_map_elem and not self.svm_label_map_elem.is_empty():
+            self.svm_label_map = \
+                pickle.loads(self.svm_label_map_elem.get_bytes())
+
+    @staticmethod
+    def _gen_param_string(params: SVM_PARAM_MAPPING_T) -> str:
+        """
+        Make a single string out of a parameters dictionary
+        """
+        return ' '.join((str(k) + ' ' + str(v)
+                         for k, v in params.items()))
+
+    def _norm_vector(self, v: np.ndarray) -> np.ndarray:
+        """
+        Class standard array normalization. Normalized along max dimension (a=0
+        for a 1D array, a=1 for a 2D array, etc.).
+
+        :param v: Vector to normalize
+
+        :return: Returns the normalized version of input array ``v``.
+        """
+        if self.normalize is not None:
+            n = numpy.linalg.norm(v, self.normalize, v.ndim - 1,
+                                  keepdims=True)
+            # replace 0's with 1's, preventing div-by-zero
+            n[n == 0.] = 1.
+            return v / n
+
+        # Normalization off
+        return v
+
+    def has_model(self) -> bool:
+        """
+        :return: If this instance currently has a model loaded. If no model is
+            present, classification of descriptors cannot happen.
+        :rtype: bool
+        """
+        return None not in (self.svm_model, self.svm_label_map)
+
+    def _train(
+        self,
+        class_examples: Mapping[Any, Iterable[DescriptorElement]]
+    ) -> None:
+        # Offset from 0 for positive class labels to use
+        # - not using label of 0 because we think libSVM wants positive labels
+        CLASS_LABEL_OFFSET = 1
+
+        # Stuff for debug reporting
+        param_debug = {'-q': ''}
+        if LOG.getEffectiveLevel() <= logging.DEBUG:
+            param_debug = {}
+
+        # Form libSVM problem input values
+        LOG.debug("Formatting problem input")
+        train_labels = []
+        train_vectors = []
+        train_group_sizes = []  # number of examples per class
+        self.svm_label_map = {}
```

Comment:
If I understand correctly, we should be able to pass labels, even as strings, directly to the scikit-learn Estimator `fit` function, and it will also record those labels in the `classes_` instance attribute, so storing those labels here is redundant.